pkg/kv/kvserver/store_split.go

```diff
@@ -19,56 +19,94 @@ import (
 	"github.com/cockroachdb/errors"
 )
 
-// splitPreApply is called when the raft command is applied. Any
-// changes to the given ReadWriter will be written atomically with the
-// split commit.
+// splitPreApplyInput contains input for the RHS replica required for
+// splitPreApply.
+type splitPreApplyInput struct {
+	// destroyed is set to true iff the RHS replica (the one with the matching
+	// ReplicaID from the SplitTrigger used to construct a splitPreApplyInput) has
+	// already been removed from the store.
+	//
+	// If the RHS replica has already been destroyed on the store, then split
+	// application entails "throwing away" the data that would have belonged to
+	// the RHS. Simply put, user data belonging to the RHS needs to be cleared.
+	destroyed bool
+	// rhsDesc is the descriptor for the post split RHS range. Set iff destroyed
+	// is false.
+	rhsDesc roachpb.RangeDescriptor
+	// initClosedTimestamp is the initial closed timestamp that the RHS replica
+	// inherits from the pre-split range. Set iff destroyed is false.
+	initClosedTimestamp hlc.Timestamp
+}
+
+// validateSplit performs invariant checks on the supplied splitTrigger and, assuming
+// they hold, returns the corresponding input that should be passed to splitPreApply.
 //
 // initClosedTS is the closed timestamp carried by the split command. It will be
-// used to initialize the new RHS range.
-func splitPreApply(
-	ctx context.Context,
-	r *Replica,
-	stateRW kvstorage.StateRW,
-	raftRW kvstorage.Raft,
-	split roachpb.SplitTrigger,
-	initClosedTS *hlc.Timestamp,
-) {
+// used to initialize the closed timestamp of the RHS replica.
+func validateSplit(
+	ctx context.Context, r *Replica, split roachpb.SplitTrigger, initClosedTS *hlc.Timestamp,
+) (splitPreApplyInput, error) {
 	// Sanity check that the store is in the split.
 	//
 	// The exception to that is if the DisableEagerReplicaRemoval testing flag is
```

Comment:
I never understood what this comment means. Feel free to ignore or fix while here if you do.